### PR TITLE
Feature/upgrade pgbouncer and pgpool 

### PR DESCRIPTION
**Pull request by plirglo**
## Postgres
- Added configurable MD5 or SHA-256 password usage for PostgreSQL
- modified core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/main.yml file to check for primary node before starting pgpool configuration (Fix for #1702) 

## PgPool:
- Configuration with pgpool version 4.2.3 (upgraded)
- libpgpool.sh scipt is mostly developed by Bitnami, i've only adde some minor changes do adjust configuration for Epiphany case - original file is located [here](https://github.com/bitnami/bitnami-docker-pgpool/blob/4.2.3-debian-10-r46/4/debian-10/rootfs/opt/bitnami/scripts/libpgpool.sh)
- some other modifications in init scripts required by newer version of pgpool
- minor required changes in pgpool.conf file

## PgBouncer:
- Configuration with pgbouncer version 1.15 (upgraded / switched to [Bitnami image](https://github.com/bitnami/bitnami-docker-pgbouncer))
- libpgbouncer.sh scipt is developed by Bitnami,  i've only adde some minor changes do adjust configuration for Epiphany case - original file is located [script](https://github.com/bitnami/bitnami-docker-pgbouncer/tree/master/1/debian-10/rootfs/opt/bitnami/scripts)
- Added pgbouncer.ini to configmap

### CODE REVIEW

**Comment by atsikham** [comment]
Target branch `feature/upgrade-postgresql` was replaced by `feature/upgrade-postgresql-13`, please change that for this PR.

File: core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2 [hunk 1]
```diff
@@ -113,6 +119,14 @@ spec:
               subPath: pgpool_postgres_password
             - name: pgpool-shared-init-dir
               mountPath: /epiphany/shared_dir
+            - name: pgpool-config-files
+              mountPath: /epiphany/bitnami/pgpool/conf/pgpool.conf
+              subPath: pgpool.conf
+            - name: pgpool-config-files
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2:125]: Please change the name of the key to not duplicate.

File: core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2 [hunk 2]
```diff
@@ -91,6 +91,9 @@ spec:
             - name: pgpool-init-scripts
               mountPath: /docker-entrypoint-initdb.d/init_pcppass.sh
               subPath: init_pcppass.sh
+            - name: pgpool-init-scripts
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2:94]: Please change the name of the key to not duplicate.

File: core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2 [hunk 3]
```diff
@@ -76,10 +76,10 @@ spec:
             - name: pgpool-passwords
               mountPath: /opt/bitnami/pgpool/secrets/pgpool_admin_password
               subPath: pgpool_admin_password
-            - name: pgpool-config-files
+            - name: pgpool-config-dir
               mountPath: /opt/bitnami/pgpool/conf/pgpool.conf
               subPath: pgpool.conf
-            - name: pgpool-config-files
+            - name: pgpool-config-dir
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/applications/templates/pgpool/04-deployment.yml.j2:82]: Please change the name of the key to not duplicate.

**Review by erzetpe — CHANGES_REQUESTED** [review]

**Review by erzetpe — DISMISSED** [review]

**Review by rafzei — DISMISSED** [review]
Just for your info: 
I see there are already the newest versions:
[Pgpool 4.2.4](https://github.com/pgpool/pgpool2/compare/V4_2_3...V4_2_4)
[Bitnami link](https://hub.docker.com/r/bitnami/pgpool/tags?page=1&ordering=last_updated)
[PGbouncer](https://github.com/pgbouncer/pgbouncer/releases/tag/pgbouncer_1_16_0)
[Bitnami link](https://hub.docker.com/r/bitnami/pgbouncer/tags?page=1&ordering=last_updated)
There is also one fix in [md5 context](https://github.com/pgbouncer/pgbouncer/commit/435d737fd4696d33100d68c6852669a2961eb7bd)

**Review by erzetpe — APPROVED** [review]

**Comment by plirglo** [comment]
> Just for your info:
> I see there are already the newest versions:
> [Pgpool 4.2.4](https://github.com/pgpool/pgpool2/compare/V4_2_3...V4_2_4)
> [Bitnami link](https://hub.docker.com/r/bitnami/pgpool/tags?page=1&ordering=last_updated)
> [PGbouncer](https://github.com/pgbouncer/pgbouncer/releases/tag/pgbouncer_1_16_0)
> [Bitnami link](https://hub.docker.com/r/bitnami/pgbouncer/tags?page=1&ordering=last_updated)
> There is also one fix in [md5 context](https://github.com/pgbouncer/pgbouncer/commit/435d737fd4696d33100d68c6852669a2961eb7bd)

Thanks, we don't use md5 encryption with plain-text passwords, so i think this should not impact our configuration

**Review by rafzei — APPROVED** [review]

**Comment by plirglo** [comment]
Applies also to task #2076

**Pull request merged**